Return request on end, just like superagent actually does.

oponder opened the pull request:
I needed to check if I set some headers correctly in my tests.

I noticed that superagent actually returns the request object in its own `end` call: https://github.com/visionmedia/superagent/blob/b91a26fb4d8acf28534ac2cdd0bb3ce4a6f17498/lib/client.js#L813

Superagent-mock does not do this. 

That means I couldn't inspect the request object I had made to ensure that some code of mine was setting headers correctly on the superagent request object.

This changes superagent-mock so that it also returns the request object for mocked requests.

Does it need tests? Let me know. :+1:

oponder commented:
Not sure what is up with travis. Can investigate and fix later, any hints welcome!

fdubost commented:
It's ok for me. You could indeed add a test to check object return by `end` :blush: 

For travis, I don't know what it's wrong, I can investigate too but not now :disappointed:

oziks commented:
@fdubost @oponder tests failed on master to :/

oponder commented:
Hey guys, not sure why the `component-as-module` things was there. I couldn't really understand its purpose, so I removed it and required superagent using plain old `require`. But I might have just circumvented something important. So please forgive me :)

oponder commented:
I also sneaked in the latest version of superagent (1.8.1) into package.json

oziks commented:
`component-as-module` is used for required `superagent` on client version. `nodeunit` execute tests into node . If you remove `component-as-module` `client.spec.js` and `server.spec.js` is the same files.

oponder commented:
Oh silly me! Was a bit too quick to dismiss. Thanks for explaining @oziks 

I will do a bit more research.

oponder commented:
I couldn't really figure out how to solve the failing test case, sorry :(

oziks commented:
@fdubost tu pense pouvoir résoudre ce problème ?

fdubost commented:
@oziks je regarde

fdubost commented:
@oziks I fix tests here #26

fdubost commented:
@oponder I have just merged #26, you can rebase your branch 😉

fdubost commented:
@oponder any news about this PR ?

oponder commented:
Ah, I missed that message a long time ago. 

I moved in the meanwhile to mocking my server with just a mock express js app that implements the endpoints and some bogus replies I want to test. So haven't looked back at this in a while. 

Just rebased. Looks to be passing.

fdubost commented:
Thanks a lot @oponder 👍
